flocker/node/agents/ebs.py

```diff
@@ -590,6 +602,11 @@ def destroy_volume(self, blockdevice_id):
                         'Failed to delete volume: {!r}'.format(blockdevice_id)
                     )
                 else:
+                    ebs_volume = self._get_ebs_volume(blockdevice_id)
+                    _wait_for_volume(ebs_volume,
+                                     start_status=u'available',
+                                     transient_status=u'deleting',
+                                     end_status='')
```

Comment:
Not directly related to this branch but `if ret_val is False:` is a bad expression.  That should be `if not ret_val:`.  And since there's an `if` and an `else` block you could make it `if ret_val:` and flip the order of the blocks for easier reading.  And `ret_val` is a pretty generic variable name.  Perhaps rename it something like `destroy_result`.